components/action-mgt/org.wso2.carbon.identity.action.execution/src/test/java/org/wso2/carbon/identity/action/execution/util/RequestFilterTest.java

```diff
@@ -85,15 +92,20 @@ public void testGetFilteredParams() {
         Mockito.when(config.getExcludedParamsInActionRequestForActionType(ActionType.PRE_ISSUE_ACCESS_TOKEN))
                 .thenReturn(excludedParams);
 
-        Map<String, String[]> params = new HashMap<>();
-        params.put("x-param-1", new String[]{"X-param-1-value"});
-        params.put("X-Param-2", new String[]{"X-Param-2-Value"});
-        params.put("x-param-3", new String[]{"X-param-3-value"});
+        List<Param> params = new ArrayList<>();
+        params.add(new Param("x-param-1", new String[]{"X-param-1-value"}));
+        params.add(new Param("X-Param-2", new String[]{"X-Param-2-Value"}));
+        params.add(new Param("x-param-3", new String[]{"X-param-3-value"}));
 
-        Map<String, String[]> filteredParams =
+        List<Param> filteredParams =
                 RequestFilter.getFilteredParams(params, ActionType.PRE_ISSUE_ACCESS_TOKEN);
         assertEquals(filteredParams.size(), 2);
-        assertEquals(filteredParams.get("x-param-3"), new String[]{"X-param-3-value"});
-        assertEquals(filteredParams.get("X-Param-2"), new String[]{"X-Param-2-Value"});
+        filteredParams.forEach(filteredParam -> {
+            if (filteredParam.getName().equalsIgnoreCase("x-param-3")) {
```

Comment:
Param is not case sensitive. You should not ignore case here. That's why this test even use different case params

Reply:
This is updated with b1aae9dc06fbd0613e3c1b079177f23ce977be61